cpp/box/Box.h

```diff
@@ -396,18 +396,26 @@ class Box
         \param dist Distances between points and query_points
         \param Nvecs The number of points to calculate distances between
     */
+
     void computeDistances(vec3<float>* query_points, vec3<float>* points, const unsigned int* query_point_indices,
-        const unsigned int* point_indices, float *dist, unsigned int Mvecs, unsigned int Nvecs) const
+        const unsigned int* point_indices, float *dist, unsigned int n_query_points, unsigned int n_points,
+        unsigned int n_query_indices, unsigned int n_point_indices) const
     {
-        if (Mvecs != Nvecs)
-            throw std::runtime_error("Inconsistent array sizes.");
-        util::forLoopWrapper(0, Nvecs, [=](size_t begin, size_t end) {
+        if (n_query_indices != n_point_indices)
+        {
+            throw std::runtime_error("Query point and point index array dimesnsions must match.");
+        }
+        util::forLoopWrapper(0, n_query_indices, [=](size_t begin, size_t end) {
             for (size_t i = begin; i < end; ++i)
             {
-                if (query_point_indices[i] > Mvecs)
-                    throw std::out_of_range("Index too big.");
-                if (point_indices[i] > Nvecs)
-                    throw std::out_of_range("Index too big.");
+                if (query_point_indices[i] >= n_query_points)
+                {
+                    throw std::out_of_range("Query point indices exceed array dimensions.");
```

Comment:
Add `#include <sstream>` at the top of this file.
```suggestion
                    std::ostringstream msg;
                    msg << "Query point index " << query_point_indices[i] << " is out of range for " << n_query_points << " provided query points."
                    throw std::out_of_range(msg.str());
```

Reply:
@klywang please implement this change, it will make error messages much more useful.